File: lib/Employee.js
//Class created to perform sql operation
import {TableQuery} from "./TableQuery.js";

 class Employee extends TableQuery
 {
     
     constructor(){
        super();
         this.select = `SELECT * FROM employee;`;
         this.insert = `INSERT INTO employee(first_name, last_name, role_id, manager_id) VALUES (?,?,?,?);`;
         this.selectById = `SELECT * FROM employee WHERE id = ?`;
         this.delete =  `DELETE FROM employee WHERE id = ?`;
         this.update = `UPDATE employee set role_id =? where id= ?;`;
         this.selectByManager = `SELECT * FROM employee WHERE manager_id= ?;`;
         this.selectAllManagers =`SELECT DISTINCT mgr.* FROM employee mgr 
                                        join employee emp ON mgr.id = emp.manager_id;`
         this.selectByDepartment= `SELECT * FROM employee emp
                                            JOIN role rl ON rl.id = emp.role_id
                                             WHERE department_id = ?;`
     }    ;

     getEmployeeByManager()
     {
        return this.selectByManager;
     }

     getManagers()
     {
        return this.selectAllManagers;
     }

     getEmployeesByDepartment()
     {
        return this.selectByDepartment;
     }
 }

export {Employee}

File: lib/Employee.cs
//Class created to perform sql operation
const TableQuery = require('./TableQuery);

 Class Employee extends TableQuery
 {
     constructor(){
         this.commonSelectFields =`SELECT emp.id as Id, emp.first_name as FirstName, emp.last_name as LastName, 
                        rl.title as 'Job Title' ,dp.name as 'Department',rl.salary,
                        mgr.first_name +" " +mgr.last_name as 'Manager'
                        FROM employee emp 
                        left join role rl on emp.role_id = rl.id 
                        left join dep dp on dp.id = rl.department_id
                        left join employee mgr on emp.manager_id = emp.id`;
         this.select = `${this.commonSelectFields};`;
         this.insert = `INSERT INTO employee(first_name, last_name, role_id, manager_id) VALUES (?,?,?,?);`;
         this.selectById = `${this.commonSelectFields} WHERE id = ?;`;
         this.delete =  `DELETE FROM department WHERE id = ?`;
     }    
 }

 module.exports = Employee;